src/source/image_source.js

```diff
@@ -140,40 +141,43 @@ class ImageSource extends Evented {
     }
 
     _setTile(tile) {
-        this.tile = tile;
+        this.tiles[tile.coord.w] = tile;
         const maxInt16 = 32767;
         const array = new RasterBoundsArray();
         array.emplaceBack(this._tileCoords[0].x, this._tileCoords[0].y, 0, 0);
         array.emplaceBack(this._tileCoords[1].x, this._tileCoords[1].y, maxInt16, 0);
         array.emplaceBack(this._tileCoords[3].x, this._tileCoords[3].y, 0, maxInt16);
         array.emplaceBack(this._tileCoords[2].x, this._tileCoords[2].y, maxInt16, maxInt16);
 
-        this.tile.buckets = {};
+        tile.buckets = {};
 
-        this.tile.boundsBuffer = Buffer.fromStructArray(array, Buffer.BufferType.VERTEX);
-        this.tile.boundsVAO = new VertexArrayObject();
+        tile.boundsBuffer = Buffer.fromStructArray(array, Buffer.BufferType.VERTEX);
+        tile.boundsVAO = new VertexArrayObject();
     }
 
     prepare() {
-        if (!this.tile || !this.image) return;
+        if (this.tiles.length === 0 || !this.image) return;
```

Comment:
`this.tiles` is an object, not an array, so I think you need `Object.keys(this.tiles).length === 0`